Lib/test/test_file.py

```diff
@@ -216,6 +216,18 @@ def testSetBufferSize(self):
         with self.assertWarnsRegex(RuntimeWarning, 'line buffering'):
             self._checkBufferSize(1)
 
+    def testDefaultBufferSize(self):
+        f = self.open(TESTFN, 'wb')
```

Comment:
Should use `with` here. Definitely unlikely that `f.write()` will fail, but if it does, this doesn't close the file.

Reply:
I think it's intentional to not use with? none of the tests in this file are using with. I think they don't expect the file class and the with context manager to be working.

Reply:
Depends which test. There's some general "test suite" setup/teardown which covers a number of cases / prevents leaks, `test_fileio` uses with, I think it's good practice. Can definitely get "you're implementing a Python and some tests working", but this is new / beyond baseline "does open work".

a fileio test that uses with:
```python
    # verify readinto
    def testReadintoByteArray(self):
        self.f.write(bytes([1, 2, 0, 255]))
        self.f.close()

        ba = bytearray(b'abcdefgh')
        with self.FileIO(TESTFN, 'r') as f:
            n = f.readinto(ba)
        self.assertEqual(ba, b'\x01\x02\x00\xffefgh')
        self.assertEqual(n, 4)
```